x-pack/plugins/security_solution/public/common/utils/global_query_string/helpers.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { parse } from 'query-string';
+import { decode, encode } from 'rison-node';
+import { SecurityPageName } from '../../../app/types';
+
+export const isDetectionsPages = (pageName: string) =>
+  pageName === SecurityPageName.alerts ||
+  pageName === SecurityPageName.rules ||
+  pageName === SecurityPageName.rulesCreate ||
+  pageName === SecurityPageName.exceptions;
+
+export const decodeRisonUrlState = <T>(value: string | undefined): T | null => {
+  try {
+    return value ? (decode(value) as unknown as T) : null;
+  } catch (error) {
+    if (error instanceof Error && error.message.startsWith('rison decoder error')) {
+      return null;
+    }
+    throw error;
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
we can probably remove this (later) by using `RisonObject | null` in the places that use this function